components/script/script_thread.rs

```diff
@@ -1581,30 +1579,11 @@ impl ScriptThread {
 
     fn collect_reports(&self, reports_chan: ReportsChan) {
         let mut path_seg = String::from("url(");
-        let mut dom_tree_size = 0;
         let mut reports = vec![];
-        // Servo uses vanilla jemalloc, which doesn't have a
-        // malloc_enclosing_size_of function.
-        let mut ops = MallocSizeOfOps::new(::servo_allocator::usable_size, None, None);
 
         for (_, document) in self.documents.borrow().iter() {
             let current_url = document.url();
-
-            for child in document.upcast::<Node>().traverse_preorder() {
-                dom_tree_size += malloc_size_of_including_self(&mut ops, &*child);
-            }
-            dom_tree_size += malloc_size_of_including_self(&mut ops, document.window());
-
-            if reports.len() > 0 {
-                path_seg.push_str(", ");
```

Comment:
We're missing the joining `, ` now. Let's do the following instead of this loop:
```
let documents = self.documents.borrow();
let urls = itertools::join(documents.values().map(|d| d.url().as_str()), ", ")
let path_seg = format!("url({})", urls);
```
(you'll need to add an `extern crate itertools` to script/lib.rs and add itertools to script/Cargo.toml, too)

Reply:
This change results in a compilation error `no method named values found for type std::cell::Ref<'_, script_thread::Documents>`. I looked at the impl for Documents, but was not able to fix the problem. I will continue working on it tomorrow.

Reply:
Oh, I made the wrong assumption about the type of self.documents. You can use `documents.iter().map(|(_, d)| d.url().as_str())` instead.

Reply:
When I put this in as is, it results in a compilation error on d.url() `borrowed value does not live long enough`. I think this is caused by as_str() returning a &str. I replaced as_str() with to_string() and it compiled and I tested it and the colon separated list of urls shows up. Please let me know if that change is ok.

Reply:
Looks good to me!